caikit/runtime/work_management/request_aborter.py

```diff
@@ -0,0 +1,107 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module helps us know when a HTTP client call is cancelled, and we need to stop or undo work
+"""
+# Standard
+from typing import Optional
+import asyncio
+import threading
+
+# Third Party
+from fastapi import Request
```

Comment:
Yeah, this needs to go in `http_server` since `fastapi` is an optional dependency. We're trying to keep all usage of these optional dependencies as isolated as possible to avoid needing to track how/when they get imported.

Reply:
In that same vein, we probably want/need to consolidate the places where `grpc` is used into a sub-module that's quarantined from proactive imports as well

Reply:
Okay that's what I thought! I'll move it over.